### PR TITLE
Add row counter for chunked parsing for accurate row reporting on errors

### PR DESCRIPTION
**Pull request by mauriciovillalobos**
Add test and long enough csv file to need more than one chunk to process

### CODE REVIEW

File: tests/node-tests.js [hunk 1]
```diff
@@ -59,6 +59,43 @@ describe('PapaParse', function() {
...
+						"message": "Too few fields: expected 3 fields but parsed 2",
+						"row": 498
+					},
+					{
+						"type": "FieldMismatch",
+						"code": "TooFewFields",
+						"message": "Too few fields: expected 3 fields but parsed 2",
+						"row": 998
+					},
+					{
```
Comment (pokoli) [inline tests/node-tests.js]: Is not enought to test with 1000 rows?

File: papaparse.js [hunk 2]
```diff
@@ -957,6 +957,7 @@
 
 		var self = this;
 		var _stepCounter = 0;	// Number of times step was called (number of rows parsed)
+		var _rowsCounter = 0;	// Total number of rows that have been parsed so far
```
Comment (pokoli) [inline papaparse.js]: I think it's better named _rowCounter

Comment (pokoli) [inline papaparse.js]: I think the total word is not requried on the comment, so it will be: _Number of rows that had been parsed so far_

**Comment by mauriciovillalobos** [comment]
Hey,

I updated the PR with your comments

I checked #441 and #432 but this doesn't seem like it's addressing any of those.
The specific issue I was having was on very large csv files, that were being loaded in chunks and the total number of rows parsed was not kept between the chunks.

I think the issue described on #441 has to do with the fact that we're using 0th index to report the row number, and not taking into account the header row if any. This could be fixed by:

- Adding a +1 to offset the 0th index
- Adding another +1 if it has a header row
- Checking if we have skipped empty rows, and take it into account (in the case of empty rows between rows with content)

But this would break current behavior, if anyone was done the offsetting themselves in the code (like I did when working with papaparse just now, I added +2 before reporting the error to my users)

My two cents, this could be fixed but maybe on another PR.

**Comment by pokoli** [comment]
Ok, now I fully understand the picture. I just asked about other issues to make see if they are related but I see is completely unrelated. Thanks for your work on this.

**Pull request merged**